x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/queries/manage_queries_modal.test.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockValues, setMockActions } from '../../../../../__mocks__';
+
+import React from 'react';
+
+import { shallow, ShallowWrapper } from 'enzyme';
+
+import { EuiButton, EuiModal } from '@elastic/eui';
+
+import { CurationQueries } from '../../components';
+
+import { ManageQueriesModal } from './';
+
+describe('ManageQueriesModal', () => {
+  const values = {
+    queries: ['hello', 'world'],
+    queriesLoading: false,
+  };
+
+  const actions = {
+    updateQueries: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(values);
+    setMockActions(actions);
+  });
+
+  describe('modal button', () => {
+    it('renders a modal toggle button', () => {
+      const wrapper = shallow(<ManageQueriesModal />);
+
+      expect(wrapper.find(EuiButton)).toHaveLength(1);
+    });
+
+    it('renders the toggle button with a loading state when queriesLoading is true', () => {
+      setMockValues({ ...values, queriesLoading: true });
+      const wrapper = shallow(<ManageQueriesModal />);
+
+      expect(wrapper.find(EuiButton).prop('isLoading')).toBe(true);
+    });
+  });
+
+  describe('modal', () => {
+    let wrapper: ShallowWrapper;
+
+    beforeEach(() => {
+      wrapper = shallow(<ManageQueriesModal />);
+      wrapper.find(EuiButton).simulate('click');
```

Comment:
It's possible to mock useState, but it's kind of a pain. The setup looks like this:

```js
jest.mock('react', () => ({
  ...(jest.requireActual('react') as object),
  useState: jest.fn(),
}));

it('some test', () => {
  const setter = jest.fn(),
  (useState as jest.Mock).returnValueOnce(['mock state', setter]);
});
```

I don't strongly object to it, but there are a couple of minor issues I have with the concept:

- Once mocked, it's not really possible to have `useState` work as expected in other tests - the setters won't actually set state as expected. This is actually distinctly unlike our logic files: we can set _default_ state for the logic files, but we can still test that state works as expected & changes based on actions. I'm hesitant to implement this for that reason; it makes it impossible to write non-mocked state tests which I do think is important to test.

- I also don't think view / React component tests necessarily need to be written the same as logic or more standard unit tests. They're kind of a different breed of "unit" tests that aren't as straightforward and cover a weird blend of UI/UX and logic. It might be worth codifying those some day for sure.